packages/web-ui/src/ds/atoms/DropzoneInput/index.tsx

```diff
@@ -18,27 +26,40 @@ export function DropzoneInput({
   placeholder,
   accept,
   multiple,
+  defaultFilename,
+  icon = 'fileUp',
+  inputSize = 'normal',
+  onChange,
   ...rest
 }: Props) {
-  const [selectedFile, setSelectedFile] = useState<File | null>(null)
+  const [filename, setFilename] = useState<string | undefined>(defaultFilename)
+  useEffect(() => setFilename(defaultFilename), [defaultFilename])
+
   const ref = useRef<HTMLInputElement>(null)
+
   const onClickZone = useCallback(() => {
     if (!ref.current) return
-
     ref.current.click()
   }, [ref])
+
   const onFileChange = useCallback(
     (files: FileList | null) => {
       const file = files?.[0]
       if (!file) return
-
-      setSelectedFile(file)
+      setFilename(file.name)
+      onChange?.(files)
     },
-    [setSelectedFile],
+    [setFilename, onChange],
   )
-  const name = selectedFile?.name
+
+  const onClearFile = useCallback(() => {
+    if (ref.current) ref.current.value = ''
+    setFilename(undefined)
+    onChange?.(null)
+  }, [ref, setFilename, onChange])
+
   return (
-    <div onClick={onClickZone}>
+    <div onClick={onClickZone} className='w-full'>
```

Comment:
hmm review this ain't breaking UI in other places where this component is used

Reply:
Reviewed, it does not break current uses 👍